src/pytest_split/plugin.py

```diff
@@ -181,6 +185,60 @@ def pytest_collection_modifyitems(
         return None
 
 
+def _reorganize_broken_up_ipynbs(group: "TestGroup", items: list) -> None:
+    """
+    Ensures that group doesn't contain partial IPy notebook cells.
+
+    ``pytest-split`` might, in principle, break up the cells of an
+    IPython notebook into different test groups, in which case the tests
+    most likely fail (for starters, libraries are imported in Cell 0, so
+    all subsequent calls to the imported libraries in the following cells
+    will raise ``NameError``).
+
+    """
+    item_node_ids = [item.nodeid for item in items]
+
+    # Deal with broken up notebooks at the beginning of the test group
+    if not group.selected or not _is_ipy_notebook(group.selected[0].nodeid):
+        return
+    first = group.selected[0].nodeid
```

Comment:
I'd change the order of these three things:
```python
first = group.selected[0].nodeid

if not group.selected or not _is_ipy_notebook(first):
        return

item_node_ids = [item.nodeid for item in items]
```

because `item_node_ids` is used only after and`group.selected[0].nodeid` deserves a variable from the beginning as it's otherwise accessed twice.

Reply:
That wouldn't work, because if `group.selected` is an empty list, `first = group.selected[0].nodeid` throws `IndexError`.

The reason the `if` statement works is that Python short circuits the `if` statement when the the first argument (`not group.selected`) evaluates to `True` (since the whole `or` clause would be true regardless of what `not_is_ipy_notbeook(...)` evaluates to).

Reply:
Oh yes, my bad, `group.selected` can be empty indeed. However, `item_node_ids = [item.nodeid for item in items]` could be done after checking for the "early exit" 🙂

Reply:
Done :)